apps/anomalist/gp_detector.py

```diff
@@ -149,17 +151,28 @@ def get_score_df(self, df: pd.DataFrame, variable_ids: List[int], variable_mappi
         df_score_long = pd.concat(results).reset_index()
 
         # Normalize the anomaly scores by mapping interval (0, 3+) to (0, 1)
-        df_score_long["anomaly_score"] = np.minimum(df_score_long["anomaly_score"] / 3, 1)
+        # df_score_long["anomaly_score"] = np.minimum(df_score_long["z"] / 3, 1)
 
-        return df_score_long
+        # Convert z-score into p-value
+        df_score_long["p_value"] = 2 * (1 - norm.cdf(np.abs(df_score_long["z"])))
+
+        # Adjust p-values for multiple testing
+        df_score_long["adj_p_value"] = df_score_long.groupby(["entity_name", "variable_id"]).p_value.transform(
+            lambda p: multipletests(p, method="fdr_bh")[1]
+        )
```

Comment:
Nice! I also implemented a more natural normalization on my branch, instead of clipping above 3. Using the CDF works well, but in practice it doesn't make much of a different for big anomalies. Everything above ~3 sigmas gets a score of 100%.
I didn't know about multipletests! Have you checked if it makes a bit difference?

Reply:
> Have you checked if it makes a bit difference?

Yeah, it does make a big difference, although I'm not really sure if using multiple hypothesis adjustments makes sense for residuals. It's not motivated by any theory, just by the need to "make the number smaller for larger samples".